packages/openapi-generator/src/document-converter.ts

```diff
@@ -0,0 +1,84 @@
+import { promises } from 'fs';
+import { parse } from 'path';
+import { OpenAPIV3 } from 'openapi-types';
+import { convert } from 'swagger2openapi';
+import { safeLoad } from 'js-yaml';
+import { errorWithCause } from '@sap-cloud-sdk/util';
+const { readFile } = promises;
+
+/**
+ * Convert an OpenApi document to ensure smooth parsing and generation thereafter.
+ * Documents are expected to be formatted as JSON, OpenApi version 3 and only have one "default" tag.
+ * @param filePath File content of the original spec.
+ * @param ouputFilePath Path to write the altered spec to.
+ */
+export async function convertOpenApiSpec(
+  filePath: string
+): Promise<OpenAPIV3.Document> {
+  let file;
+  try {
+    file = await parseFileAsJson(filePath);
+  } catch (err) {
+    throw errorWithCause('Could not parse OpenApi specification.', err);
+  }
+  const openApiDocument = await convertDocToOpenApiV3(file);
+  return convertDocToGlobalTag(openApiDocument);
+}
+
+/**
+ * Parse a JSON or YAML file and return it as JSON.
+ * @param filePath Path to the file
+ * @returns JSON representation of the given file.
+ */
+export async function parseFileAsJson(
+  filePath: string
+): Promise<Record<string, any>> {
+  const fileContent = await readFile(filePath, 'utf8');
+  const extension = parse(filePath).ext;
+  if (extension === '.json') {
+    return JSON.parse(fileContent);
+  }
+  if (['.yaml', '.yml'].includes(extension)) {
```

Comment:
question: will this handle capital case like `.YAML` and the json above?

Reply:
No, I can add it, even though this is not standard ;)

Reply:
Done.